tools/pkg/engine_build_configs/bin/check.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io' as io;
+
+import 'package:engine_build_configs/engine_build_configs.dart';
+import 'package:engine_repo_tools/engine_repo_tools.dart';
+import 'package:path/path.dart' as p;
+
+// Usage:
+// $ dart bin/check.dart [/path/to/engine/src]
+
+void main(List<String> args) {
+  late final String? engineSrcPath;
+  if (args.isNotEmpty) {
+    engineSrcPath = args[0];
+  } else {
+    engineSrcPath = null;
+  }
+
+  // Find the engine repo.
+  late final Engine engine;
```

Comment:
Same here I believe.

Please also apply elsewhere in the code (many but not all of your late usages elsewhere in the PR).